### PR TITLE
release/v2.1.0-preview.1.0.0

### PR DESCRIPTION
**Pull request by andreise**
- Add F#-like comparison
- release/v2.1.0-preview.1.0.0

### CODE REVIEW

File: src/core-unit/Unit/Unit/Unit.Comparison.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
+﻿namespace System;
+
+partial struct Unit
+{
+    public int CompareTo(Unit other) => 0;
```
Comment (pmosk) [inline src/core-unit/Unit/Unit/Unit.Comparison.cs]: Here and next. Shell we use "default" instead constant zero?

File: src/core-unit/Unit/Unit/Unit.Comparison.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+    public int CompareTo(object? obj) => obj switch
+    {
+        null => 1,
+        Unit => 0,
+        _ => throw new ArgumentException("The object is not the Unit.", nameof(obj))
+    };
+
+    public static int Compare(Unit left, Unit right) => (left, right) switch
```
Comment (pmosk) [inline src/core-unit/Unit/Unit/Unit.Comparison.cs]: Will it be faster if we just return zero?

Comment (andreise) [reply src/core-unit/Unit/Unit/Unit.Comparison.cs]: Done: redundant switch is removed.

File: src/core-unit/Unit/Unit/Unit.Comparison.cs (hunk 1 above)
Comment (andreise) [reply src/core-unit/Unit/Unit/Unit.Comparison.cs]: Done: default is used instead of literal zero.

**Review by pmosk — APPROVED** [review]

**Pull request merged**